exercises/state-of-tic-tac-toe/description.md

```diff
@@ -0,0 +1,69 @@
+# Description
+
+In this exercise, you're going to implement a program that determines the state of a [tic-tac-toe](https://en.wikipedia.org/wiki/Tic-tac-toe) game.
+(_You may also know the game as "noughts and crosses" or "Xs and Os"._)
+
+The games is played on a 3×3 grid.
+Players take turns to place `X`s and `O`s on the grid.
+The game ends when one player has won by placing three of marks in a row, column, or along a diagonal of the grid, or when the entire grid is filled up.
+
+In this exercise, we will assume that `X` starts.
+
+It's your job to determine which state a given game is in.
+
+There are 3 potential game states:
+
+- The game is **ongoing**.
+- The game ended in a **draw**.
+- The game ended in a **win**.
+
+If the given board is invalid, throw an appropriate error.
+
+If a board meets the following conditions, it is invalid:
+
+- The players took turns out of order (remember that `X` starts).
+- The game was played after it already ended.
+
```

Comment:
How do we test for out of order?  Are there tests for out of order?  I only noticed snapshots and any snapshot that has equal or one less "O" than "X" can not indicate play order on its own.  I also do not see an example in "Examples" that shows what this might look like.

Reply:
>  I only noticed snapshots and any snapshot that has equal or one less "O" than "X" can not indicate play order on its own.

A board that has more Os than Xs is impossible to reach when the right turn order is followed. We only care if the snapshot is a valid board, i.e. reachable through legal turns.

Reply:
So nothing to clarify about this in the description?  Line 24 mentions it, and I looked through and did not notice an example of an "out of order" state.  But does this matter?  Are we testing for this?  (Yes, a board that has more than 2 less "O"'s than "X" would an invalid state, etc.

Reply:
There are two tests for it towards the end of the file. I'll add an example.

Reply:
@kotp does [2f1e126](https://github.com/exercism/problem-specifications/pull/1991/commits/2f1e126d1ceb9bb1901fbfc94e4b29b6e5d0bb77) clear things up?

Reply:
Ah, yes, that works.